app/js/PersistorHelper.js

```diff
@@ -131,6 +136,15 @@ function wrapError(error, message, params, ErrorType) {
       message: 'no such file',
       info: params
     }).withCause(error)
+  } else if (
+    error instanceof UpstreamError ||
+    parseInt(error.code) >= 500 ||
+    (error.message && error.message.match(/^Cannot parse response as JSON.*/))
+  ) {
+    return new UpstreamError({
+      message: 'internal error from GCS',
```

Comment:
So this actually `GCSUpstreamError`?

Reply:
No, I should make the error more generic. (If S3 returns a 5xx error code, this logic will kick in as well)

Reply:
> (If S3 returns a 5xx error code, this logic will kick in as well)

Can you adjust the message then, to not include `GCS`?

Reply:
Yeah, that's what I meant :-)

Reply:
I should read more closely :D